packages/x-data-grid/src/hooks/features/filter/useGridFilter.tsx

```diff
@@ -107,7 +107,7 @@ export const useGridFilter = (
 
   const updateFilteredRows = React.useCallback(() => {
     apiRef.current.setState((state) => {
-      const filterModel = gridFilterModelSelector(state, apiRef.current.instanceId);
+      const filterModel = gridFilterModelSelector(state, undefined, apiRef.current.instanceId);
```

Comment:
This is ugly :| We could instead duplicate `.instanceId` on the state and remove those args, though the refactor might be useless if we do #11440.